azurelinuxagent/ga/update.py

```diff
@@ -373,6 +385,9 @@ def run(self, debug=False):
                 time.sleep(self._goal_state_period)
 
         except ExitException as exitException:
+            if exitException.add_event:
+                add_event(AGENT_NAME, op=WALAEventOperation.AgentUpgrade, version=CURRENT_VERSION, is_success=True,
```

Comment:
it isn't clear to me what guarantees that an exception with the `add_event` flag set would come from an `AgentUpgrade` operation, besides just that each and every `raise ExitException` has `add_event` set if and only if it comes from an `AgentUpgrade` operation.

Could we either: 
* make this code more general, i.e. capture _all_ of the event information in the exception object
* make the name of the flag more specific, i.e. change it from `add_event` to something like `caused_by_upgrade` or something

Just seems like there's currently a mismatch in specificity here.

Reply:
Sorry for the confusion, the expectation is not that the ExitException has to come from an AgentUpgrade scenario to send an event, its just that we should not send an event for a very specific case (this one `Agent {0} is an orphan -- exiting`) and add an event for all the other cases :)
Hope that clears it up!